geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreFactoryImpl.java

```diff
@@ -141,10 +141,9 @@ DiskStoreImpl createOwnedByRegion(String name, boolean isOwnedByPR,
   @Override
   public DiskStore create(String name) {
     this.attrs.name = name;
-    // As a simple fix for 41290, only allow one DiskStore to be created
-    // at a time per cache by syncing on the cache.
     DiskStore result;
-    synchronized (this.cache) {
+    try {
+      this.cache.lockDiskStore(name);
```

Comment:
Does this needs to be before "try"? if lock throws exception without locking we may be trying to unlock it...

Reply:
The `unlock` in the finally block does nothing, if the `lock` fails. The `unlock` is essentially a `CountDownLatch.countDown()`. If the current count is greater than zero then it is decremented. If the current count equals zero then nothing happens.